### DISCUSSION TITLE
Removing only Shapes in PDF page

### DISCUSSION THREAD

**Discussion by jjfang** [pymupdf/PyMuPDF#2308] [Looking for help] [answered]
Hello, I have a PDF document with pages containing images, text, and shapes.

I was wondering if there is a way to remove only certain shapes retrieved via the Page.get_drawings() function, while leaving all the other objects in the page the same. Something like the Page.delete_image() function, but for shapes instead.

Thanks!

**Comment by JorjMcKie** [comment, marked as answer]
Hi, no that is not supported, sorry.

**Reply by jjfang** [reply]
do you know if this will be in the roadmap somewhere down the line?

**Reply by JorjMcKie** [reply]
no, there are no such plans.

**Reply by jjfang** [reply]
thanks for the quick reply!

**Reply by aqz236** [reply]
Hello, PyMuPDF is very powerful and I thank you for creating it! ...Is it now supported to delete vector images? I recently encountered this need at work.

**Reply by JorjMcKie** [reply]
@aqz236 it still is not supported. **_But:_** there are plans to support (= include) removal of vector graphics by way of **_redaction annotations_**. In a similar way as we do images.
Probably, we will however **_not_** support **_partial_** removal, only either all or nothing.

**Reply by aqz236** [reply]
thanks for the quick reply!
